### DISCUSSION TITLE
Inferring cost of trace from metadata field if optionally provided

### DISCUSSION THREAD

**Discussion by sumba101** [langfuse/langfuse#991] [Ideas]
Although cost is tracked within Langfuse, developers tend to track it with alot of other services (for example, in their LLM provider proxy layer) and would have a need to track custom cases (such as self hosting or models not covered by Langfuse model-cost tracker)

This is an issue i have faced personally and is leading to completely inaccurate measures on my Langfuse dashboard, which reduces my Langfuse dashboard to have 4-5 useless graphs or inaccurate information 

The solution?
Reading response cost for a generation from a 'response_cost' field from metadata, if present. The field shall act as the indicator for cost of tokens for that trace. 

Benefits?
Allows for High degree of flexibility with usage of Langfuse and opens up a plethora of intergration of other services with Langfuse

Please vote to support this feature

**Comment by marcklingen** [comment]
Thanks for sharing this @sumba101 

A major change to the cost/usage api is currently in progress which will make it very simple to ingest token usage and/or USD cost of llm calls, addressing exactly this. I'll ping you once it's released.

**Reply by sumba101** [reply]
thank you for the response. Excited for the release <3

**Comment by marcklingen** [comment]
@sumba101 We've just released [v2.0.0](https://github.com/langfuse/langfuse/releases/tag/v2.0.0) which optionally allows to ingest costs for each generation object. Hope you like it. Let me know if you have any feedback/thoughts.
